src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -3361,6 +3361,15 @@ public override void PostVisit(DottedNameNode node)
                         return;
                     }
 
+                    bool isBindingPropertyFunctionPropertyOrParameter = template.IsComponent &&
+                        (_txb.Document?.Properties?.EnabledFeatures?.IsEnhancedComponentFunctionPropertyEnabled ?? false) &&
+                        ((_txb.Property?.IsScopeVariable ?? false) || (_txb.Property?.IsScopedProperty ?? false));
```

Comment:
Did you validate this with Action and Event properties? Both also have scope parameters.

Reply:
This will likely broke those property scenarios. You need to ensure that current property is function property first.